src/Series.cpp

```diff
@@ -564,6 +564,10 @@ Series::flushFileBased()
         bool allDirty = dirty;
         for( auto& i : iterations )
         {
+            if ( *i.second.skipFlush )
+            {
+                continue;
```

Comment:
shouldn't we throw on this in case a users continues to issue API calls on an (accidentally) closed iteration?

Reply:
The flushing procedures haven't been written with the possibility in mind that an iteration might not be present any longer. Flushing an iteration will e.g. always open the corresponding file, irregardless of whether it is still there or not.

Reply:
This looks like something that will be very hard to debug. Maybe I missed this: When would a closed and flushed iteration be flushed again?

Reply:
>When would a closed and flushed iteration be flushed again?

The idea behind continuing with the next iteration in that case is exactly to *not* flush it again. I've added sanity checks now to check whether the iteration hasnt been wrongly accessed and we can safely skip it.